src/pyk/proof/exploration.py

```diff
@@ -0,0 +1,359 @@
+from __future__ import annotations
+
+import json
+import logging
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from pyk.kore.rpc import LogEntry
+
+from ..kast.manip import flatten_label
+from ..kcfg import KCFG
+from ..prelude.ml import mlAnd, mlTop
+from ..utils import ensure_dir_path, hash_str, shorten_hashes
+from .equality import ProofSummary
+from .proof import CompositeSummary, Proof, ProofStatus, Prover
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable, Mapping
+    from pathlib import Path
+    from typing import Any, Final, TypeVar
+
+    from ..kast.inner import KInner
+    from ..kast.outer import KClaim
+    from ..kcfg import KCFGExplore
+    from ..kcfg.kcfg import NodeIdLike
+
+    T = TypeVar('T', bound='Proof')
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+class ExplorationProof(Proof):
+    kcfg: KCFG
+    init: int
```

Comment:
```suggestion
    init: NodeIdLike
```

Any reason to use `int` here?

Reply:
The identifiers of KCFG nodes are `int`s. By having an `int` here, we minimise the calls to `_resolve` to only the class constructor. It also prevents malformed proofs from being constructed, for example, proofs that provide an `init` that is not in the given `kcfg`.

From what I've noticed, `NodeIdLike` is only used for input arguments of functions and is almost always immediately resolved. The only place where a function returns `NodeIdLike` is the `kcfg.prune` function, which returns `list[NodeIdLike]`, but it actually returns `list[int]`, and should probably be typed with this return type unless there is really a need for `NodeIdLike` to be there.